src/idl_parser.cpp

```diff
@@ -3153,13 +3153,29 @@ CheckedError Parser::ParseRoot(const char *source, const char **include_paths,
   return NoError();
 }
 
+// Generate a unique hash for a file based on its name and contents (if any).
+static uint64_t HashFile(const char *source_filename, const char *source) {
+  auto stripped_source_fillename = StripPath(source_filename);
+  auto hash = HashFnv1a<uint64_t>(stripped_source_fillename.c_str());
+  if (source && *source) hash ^= HashFnv1a<uint64_t>(source);
+
+  return hash;
+}
+
 CheckedError Parser::DoParse(const char *source, const char **include_paths,
                              const char *source_filename,
                              const char *include_filename) {
+  uint64_t source_hash = 0;
   if (source_filename) {
-    if (included_files_.find(source_filename) == included_files_.end()) {
-      included_files_[source_filename] =
-          include_filename ? include_filename : "";
+    // If the file is in-memory, don't include its contents in the hash as we
+    // won't be able to load them later.
+    if (FileExists(source_filename))
+      source_hash = HashFile(source_filename, source);
+    else
+      source_hash = HashFile(source_filename, nullptr);
+
```

Comment:
I don't understand how it works.
Why parse should continue pressing if a schema has `include filename` directive but the file doesn't exist?
Can you give an example that covers both paths of this `if (FileExists(source_filename))`?

Reply:
see [other reply](https://github.com/google/flatbuffers/pull/6434/files#r580745014)